src/Fields/Fields.jl

```diff
@@ -465,7 +465,7 @@ ClimaCore.Fields.truncate_printing_field_types() = true
 =#
 truncate_printing_field_types() = false
 
-function Base.show(io::IO, ::Type{T}) where {T <: Fields.Field}
+function Base.show(io::IOContext, ::Type{T}) where {T <: Fields.Field}
```

Comment:
does this work? why not my suggestion above?

Reply:
This works (truncates the stack traces), but invalidates ~150 methods instead of ~1k.

The suggestion above (`Base.show(io::IO, T::Type{Fields.Field{V, S}) where {V,S}`) still invalidates ~1k methods

Reply:
This was all tested with the above MWE (which seems to well reproduce what we're seeing).

Reply:
I'd like to figure out a bit more what is being invalidated

Reply:
I think we're invalidating https://github.com/JuliaLang/julia/blob/36034abf26062acad4af9dcec7c4fc53b260dbb4/base/show.jl#L881